pyo3-macros-backend/src/method.rs

```diff
@@ -604,107 +603,123 @@ impl<'a> FnSpec<'a> {
     }
 }
 
-#[derive(Debug)]
-struct MethodAttributes {
-    ty: Option<MethodTypeAttribute>,
-    python_name: Option<syn::Ident>,
+enum MethodTypeAttribute {
+    New(Span),
+    ClassMethod(Span),
+    StaticMethod(Span),
+    Getter(Span, Option<Ident>),
+    Setter(Span, Option<Ident>),
+    ClassAttribute(Span),
 }
```

Comment:
I think most of our structs and enums carry around an `Ident`, not just the span, like...
```suggestion
enum MethodTypeAttribute {
    New(Ident),
    ...
}
```

Also ...
```rust
fn ident(&self) -> &'static str {
```

...it seems weird to not return an `Ident` here.

Reply:
We don't really need the idents again, just useful in error messages. I've replaced `fn ident` with a `Display` implementation, which is hopefully a bit less weird.